File: go-version/leetcode/LC2918/min_sum.go
package LC2918

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/5/10 下午8:39
 * @FilePath: min_sum
 * @Description: 最小和，遍历 + 分类
 */

func minSum(nums1, nums2 []int) int64 {
	sum1, cntZero1 := calculate(nums1)
	sum2, cntZero2 := calculate(nums2)
	if sum1 > sum2 && cntZero2 == 0 {
		return -1
	}
	if sum2 > sum1 && cntZero1 == 0 {
		return -1
	}
	return max(sum1, sum2)
}

func calculate(nums []int) (int64, int64) {
	var sum, cnt0 int64
	for _, num := range nums {
		sum += int64(num)
		if num == 0 {
			sum++
			cnt0++
		}
	}
	return sum, cnt0
}

File: python-version/LC2918/min_sum.py
class Solution:
    def minSum(self, nums1: list[int], nums2: list[int]) -> int:
        res1, res2 = self.calculate(nums1), self.calculate(nums2)
        if res1[0] > res2[0] and res2[1] == 0:
            return -1
        if res2[0] > res1[0] and res1[1] == 0:
            return -1
        return max(res1[0], res2[0])

    @staticmethod
    def calculate(nums: list[int]) -> list[int]:
        sm, cnt0 = 0, 0
        for num in nums:
            sm += num if num > 0 else 1
            cnt0 += 1 if num == 0 else 0
        return [sm, cnt0]
